coordinator/src/services/translation/request/agent-unattended.ts

```diff
@@ -11,49 +11,64 @@ import {
+import {getAgentPersonPersonIdForAuthor} from "./practitioner"
 
-export async function createAuthorFromAuthenticatedUserDetails(
-  organizationCode: string,
-  headers: Hapi.Util.Dictionary<string>,
-  telecom: fhir.ContactPoint,
-  logger: pino.Logger
-): Promise<hl7V3.Author> {
-  const agentPerson = await createAgentPersonFromAuthenticatedUserDetails(organizationCode, headers, telecom, logger)
+export function createAuthor(
+  practitionerRole: fhir.PractitionerRole,
+  organization: fhir.Organization
+): hl7V3.Author {
+  const agentPerson = createAgentPersonUsingPractitionerRoleAndOrganization(practitionerRole, organization)
   const author = new hl7V3.Author()
   author.AgentPerson = agentPerson
   return author
 }
 
-export async function createAgentPersonFromAuthenticatedUserDetails(
-  organizationCode: string,
-  headers: Hapi.Util.Dictionary<string>,
-  telecom: fhir.ContactPoint,
-  logger: pino.Logger
-): Promise<hl7V3.AgentPerson> {
-
-  const sdsRoleProfileId = getSdsRoleProfileId(headers)
-  const sdsJobRoleCode = getRoleCode(headers)
-  const sdsUserUniqueId = getSdsUserUniqueId(headers)
-  const name = getUserName(headers)
+export function createAgentPersonUsingPractitionerRoleAndOrganization(
+  practitionerRole: fhir.PractitionerRole,
+  organization: fhir.Organization,
+): hl7V3.AgentPerson {
+  const agentPerson = new hl7V3.AgentPerson()
 
-  return createAgentPerson(
-    organizationCode,
-    sdsRoleProfileId,
-    sdsJobRoleCode,
-    sdsUserUniqueId,
-    name,
-    telecom,
-    logger
+  const sdsId = getIdentifierValueForSystem(
+    practitionerRole.identifier,
+    "https://fhir.nhs.uk/Id/sds-role-profile-id",
+    'Parameters.parameter("agent").resource.identifier'
   )
+  agentPerson.id = new hl7V3.SdsRoleProfileIdentifier(sdsId)
+
+  const sdsRoleCode = getCodeableConceptCodingForSystem(
+    practitionerRole.code,
+    "https://fhir.hl7.org.uk/CodeSystem/UKCore-SDSJobRoleName",
+    'Parameters.parameter("agent").resource.code'
+  ).code
+
+  agentPerson.code = new hl7V3.SdsJobRoleCode(sdsRoleCode)
+
+  agentPerson.telecom = [convertTelecom(practitionerRole.telecom[0], "")]
+
+  agentPerson.agentPerson = createAgentPersonPersonUsingPractitionerRole(practitionerRole)
+
+  agentPerson.representedOrganization = convertOrganization(organization, practitionerRole.telecom[0])
+
+  return agentPerson
 }
 
-export async function createAuthorFromPractitionerRole(
-  practitionerRole: fhir.PractitionerRole, logger: pino.Logger,
-): Promise<hl7V3.Author> {
-  const agentPerson = await createAgentPersonFromPractitionerRole(practitionerRole, logger)
-  const author = new hl7V3.Author()
-  author.AgentPerson = agentPerson
-  return author
+export function createAgentPersonPersonUsingPractitionerRole(
+  practitionerRole: fhir.PractitionerRole,
+): hl7V3.AgentPersonPerson {
+
+  const practitioner = practitionerRole.practitioner as fhir.IdentifierReference<fhir.Practitioner>
```

Comment:
I believe we need to specify the `as` here because the `practitionerRole.practitioner` can either be a type of `Reference `or an `IdentifierReference`, so when attempting to resolve line 62 it was throwing an error due to the possibility of it possibly being a Reference type, if my memory serves right.

Reply:
just wondering if we could use a type guard here then rather than `as` e.g.

```
const practitioner = practitionerRole.practitioner
if (!isReference(practitoner)) {
 throw new Error( wrong type)
}
```